pkg/lorabasics/patches/0001-smtc_ral-ral_defs-remove-const-from-radio_type-tcxo_.patch

```diff
@@ -0,0 +1,27 @@
+From 73580e23be59f585a0d551002f4fcb7cd5126561 Mon Sep 17 00:00:00 2001
+From: Francisco Molina <femolina@uc.cl>
+Date: Mon, 4 Apr 2022 08:26:40 +0200
+Subject: [PATCH] smtc_ral/ral_defs: remove const from radio_type/tcxo_cfg defs
+
+---
+ smtc_ral/src/ral_defs.h | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/smtc_ral/src/ral_defs.h b/smtc_ral/src/ral_defs.h
+index a7abc25..bf1ce6f 100644
+--- a/smtc_ral/src/ral_defs.h
++++ b/smtc_ral/src/ral_defs.h
+@@ -97,8 +97,8 @@ typedef struct ral_tcxo_conf_s
+ typedef struct ral_s
+ {
+     const void*             context;
+-    const ral_radio_types_t radio_type;
+-    const ral_tcxo_cfg_t    tcxo_cfg;
++    ral_radio_types_t radio_type;
```

Comment:
they are modified in the setup function : 

```
void sx1280_setup(sx1280_t *dev, const sx1280_params_t *params, uint8_t index)
{
    dev->ral.context = dev;
    dev->ral.radio_type = RAL_RADIO_SX1280;
    dev->ral.tcxo_cfg.tcxo_ctrl_mode = RAL_TCXO_NONE;
```

Reply:
It could be done as you suggest but it would have users of `sx1280_t` having to declare the `ral_t` structure (and replace the structure field with a pointer) eg:

```
ral_t ral= { .context = NULL,
                 .radio_type = RAL_RADIO_SX1280,
                .tcxo_cfg = {  .tcxo_ctrl_mode = RAL_TCXO_NONE} };
sx128_t dev = { .ral = &ral };
```
or through static initialization (we could provide a static initializer macro for that).

```
sx1280_t sx1280 ={ .ral.radio_type = RAL_RADIO_SX1280, .ral.tcxo_cfg.tcxo_ctrl_mode = RAL_TCXO_NONE};
```

What I see as an issue here is that the user needs to now handle this internal `ral` stuff, sure we could handle it in auto_init, but that just hides it IMO.

What do you think @aabadie? I can suggest a patch for the second alternative (the first I really don't like too much)